src/accelerate/accelerator.py

```diff
@@ -2243,14 +2243,19 @@ def gather(self, tensor):
         """
         return gather(tensor)
 
-    def gather_for_metrics(self, input_data):
+    def gather_for_metrics(self, input_data, use_gather_object=False):
         """
         Gathers `input_data` and potentially drops duplicates in the last batch if on a distributed system. Should be
         used for gathering the inputs and targets for metric calculation.
 
         Args:
             input (`torch.Tensor`, `object`, a nested tuple/list/dictionary of `torch.Tensor`, or a nested tuple/list/dictionary of `object`):
                 The tensors or objects for calculating metrics across all processes
+            use_gather_object(`bool`):
+                Whether to use or not gather_object instead of gather. This flag can be useful for gathering tensors
```

Comment:
```suggestion
                Whether to forcibly use gather_object instead of gather (which is already done if all objects passed do not contain tensors). This flag can be useful for gathering tensors
```

Reply:
Was not very clear to me unless I read `gather_object` description, maybe worth including it also and/or adding an example?
`Recursively gather object in a nested list/tuple/dictionary of objects from all devices.`